optaplanner-core/src/main/java/org/optaplanner/core/impl/solver/scope/SolverScope.java

```diff
@@ -35,6 +37,8 @@
  */
 public class SolverScope<Solution_> {
 
+    private final AbstractSolver<Solution_> solver;
+
```

Comment:
veto -1 (hard)

Scopes are data objects and cannot reference services. Oh wait, it does now ScoreDirector and Random.

Well, still Solver knows SolverScope, but not visa versa. Let's discuss?